include/engine/api/waypoint.hpp

```diff
@@ -0,0 +1,22 @@
+#ifndef OSRM_ENGINE_API_WAYPOINT_HPP
+#define OSRM_ENGINE_API_WAYPOINT_HPP
+
+#include "util/coordinate.hpp"
+
+namespace osrm
+{
+namespace engine
+{
+namespace api
+{
+struct Waypoint
+{
+    double distance;
+    const char *name;
```

Comment:
Why not a string? Or if you want zero-copy onto our datafacades why not a string view/ref?